File: common/src/main/java/dev/number6/db/model/converter/SentimentTotalConverter.java
package dev.number6.db.model.converter;

import dev.number6.db.model.SentimentTotal;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collection;

public class SentimentTotalConverter implements DynamoDBTypeConverter<String, Collection<SentimentTotal>> {

    private final Gson gson = new Gson();

    @Override
    public String convert(Collection<SentimentTotal> totals) {
        return gson.toJson(totals);
    }

    @Override
    public Collection<SentimentTotal> unconvert(String object) {
        Type type = new TypeToken<Collection<SentimentTotal>>() {
        }.getType();
        return gson.fromJson(object, type);
    }
}

File: comprehend/src/main/kotlin/dev/number6/db/model/converter/SentimentTotalConverter.kt
package dev.number6.db.model.converter

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import dev.number6.db.model.SentimentTotal

class SentimentTotalConverter : DynamoDBTypeConverter<String, Collection<SentimentTotal>> {
    private val gson = Gson()
    override fun convert(totals: Collection<SentimentTotal>): String {
        return gson.toJson(totals)
    }

    override fun unconvert(`object`: String): Collection<SentimentTotal> {
        val type = object : TypeToken<Collection<SentimentTotal>>() {}.type
        return gson.fromJson(`object`, type)
    }
}